### PR TITLE
Add Visual Basic support 

### PR DESCRIPTION
**Pull request by cslong**
## Related issue

Closes: #ISSUE_NUMBER_HERE

## Description
* Describe the changes in this pull request

## Supplemental testing
Describe any testing done in addition to existing unit tests

## Additional context
Please provide any additional information related to this PR

---
*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

### CODE REVIEW

File: src/CTA.Rules.Actions/VisualBasic/InterfaceActions.cs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+            }
+            return AddAttribute;
+        }
+        public Func<SyntaxGenerator, InterfaceBlockSyntax, InterfaceBlockSyntax> GetAddCommentAction(string comment)
+        {
+            InterfaceBlockSyntax AddComment(SyntaxGenerator syntaxGenerator, InterfaceBlockSyntax node)
+            {
+                var currentTrivia = node.GetLeadingTrivia();
+                //TODO see if this will lead NPE    
+                currentTrivia = currentTrivia.Add(SyntaxFactory.SyntaxTrivia(SyntaxKind.CommentTrivia,
```
Comment (saikiranakula-amzn) [inline src/CTA.Rules.Actions/VisualBasic/InterfaceActions.cs:77]: ensure the NPE exceptions might not occur

File: src/CTA.Rules.Actions/VisualBasicActionsLoader.cs [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+        _identifierNameActions = new List <MethodInfo>();
+        _attributeActions = new List <MethodInfo>();
+        _attributeListActions = new List <MethodInfo>();
+        _typeBlockActions = new List <MethodInfo>();
+        _elementAccessActions = new List <MethodInfo>();
+        _expressionActions = new List <MethodInfo>();
+        _interfaceActions = new List <MethodInfo>();
+        _methodBlockActions = new List <MethodInfo>();
+        _objectCreationExpressionActions = new List<MethodInfo>();
+        memberAccessActions = new List<MethodInfo>();
```
Comment (saikiranakula-amzn) [inline src/CTA.Rules.Actions/VisualBasicActionsLoader.cs:61]: ensure the naming convention follow the pattern "_" before variable name

**Comment by saikiranakula-amzn** [comment]
did a very high level review, ensure that NPE comment is properly addressed and all the existing test works fine without any issues

File: tst/CTA.Rules.Test/UtilsTest.cs [hunk 3]
```diff
@@ -112,5 +113,12 @@ public void ThreadSafeExportStringToFile_Permits_Concurrent_Processes()
             // Confirm that text was written to file
             Assert.AreEqual(content, File.ReadAllText(filePath));
         }
+
+        [Test]
+        public void Test_Is_VisualBasic_Project()
+        {
+            Assert.IsFalse(VisualBasicUtils.IsVisualBasicProject("test.csproj"));
+            Assert.IsTrue(VisualBasicUtils.IsVisualBasicProject("C://user/john/repos/test.vbproj"));
```
Comment (jonlouie) [inline tst/CTA.Rules.Test/UtilsTest.cs:121]: Add test case where `vbproj` is in the path or name of the file (not in the extension)

Comment (cslong) [reply tst/CTA.Rules.Test/UtilsTest.cs:121]: done

File: src/CTA.Rules.Actions/VisualBasic/ElementAccessActions.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+                SyntaxTriviaList currentTrivia = node.GetLeadingTrivia();
+                currentTrivia = currentTrivia.Insert(0, SyntaxFactory.SyntaxTrivia(SyntaxKind.CommentTrivia, string.Format(Constants.VbCommentFormat, comment)));
+                node = node.WithLeadingTrivia(currentTrivia).NormalizeWhitespace();
+                return node;
+            }
+            return AddComment;
+        }
+        public Func<SyntaxGenerator, MemberAccessExpressionSyntax, MemberAccessExpressionSyntax> GetReplaceElementAccessAction(string newExpression)
+        {
+            // todo: for VB rules that invoke this action probably need to be converted to invocation expressions because
```
[the remark this answers is not in the document]
Comment (cslong) [reply src/CTA.Rules.Actions/VisualBasic/ElementAccessActions.cs:28]: removed the comment

**Review by jonlouie — APPROVED** [review]

File: src/CTA.Rules.Metrics/Models/CTAMetric.cs [hunk 5]
```diff
@@ -1,10 +1,25 @@
...
+        
+        protected static string GetLanguage(string projectPath)
+        {
+            return projectPath.EndsWith(".csproj",
+                StringComparison.InvariantCultureIgnoreCase)
+                ? "csharp"
+                : projectPath.EndsWith(".vbproj",
+                    StringComparison.InvariantCultureIgnoreCase)
+                    ? "visualbasic"
+                    : "unknown";
```
Comment (jonlouie) [inline src/CTA.Rules.Metrics/Models/CTAMetric.cs:22]: Not a blocker, but in the future, avoid nested ternaries for readability.

**Review by jonlouie — APPROVED** [review]

**Review by saikiranakula-amzn — APPROVED** [review]

**Pull request merged**